src/serializer/ResidualHeapSerializer.js

```diff
@@ -713,7 +713,7 @@ export class ResidualHeapSerializer {
         let error = new CompilerDiagnostic(
```

Comment:
The comment above and the help text for the diagnostic does not make it clear whether this is due to a user error (or the incompleteness of Prepack) or because there is a bug in Prepack. Since the latter case should always be checked via an invariant, I assume it is the former. Please make this a bit more obvious by clarifying the comment and adding some more context to the help text.

Reply:
It's due to incompleteness of ES2015 Prepack currently, the wiki explains more on this. Once the edge cases are handled, I'll move this to an invariant (as per the PR that explained this when merged).